### ISSUE TITLE
Incremental repair doesn't work

### ISSUE THREAD

**Issue by alekiv** [thelastpickle/cassandra-reaper#71]
Tried to use incremental repairs with latest reaper snapshot but sstables didn't get marked with "repaired at" timestamp. And repair has finished too quickly (6min), although previous non incremental repair took ~4 hours.

Reaper log:
```
INFO   [2017-03-31 18:51:20,979] [dw-48 - POST /repair_run?clusterName=test1&keyspace=testks&owner=user&repairParallelism=PARALLEL&incrementalRepair=true] c.s.r.r.RepairRunResource - add repair run called with: clusterName = Optional.of(test1), keyspace = Optional.of(testks), tables = Optional.absent(), owner = Optional.of(user), cause = Optional.absent(), segmentCount = Optional.absent(), repairParallelism = Optional.of(PARALLEL), intensity = Optional.absent(), incrementalRepair = Optional.of(true)
...
INFO   [2017-03-31 18:57:11,194] [test1:1] c.s.r.s.RepairRunner - Next segment to run : 12
INFO   [2017-03-31 18:57:28,743] [test1:1:12] c.s.r.s.SegmentRunner - It is ok to repair segment '12' on repair run with id '1'
INFO   [2017-03-31 18:57:28,745] [test1:1:12] c.s.r.c.JmxProxy - Triggering repair of range (5700662054527062566,5707996255960520728] for keyspace "testks" on host 192.168.0.10, with repair parallelism parallel, in cluster with Cassandra version '3.0.11' (can use DATACENTER_AWARE 'true'), for column families: []
INFO   [2017-03-31 18:57:28,746] [test1:1:12] c.s.r.s.SegmentRunner - Repair for segment 12 started, status wait will timeout in 18000000 millis
INFO   [2017-03-31 18:57:31,790] [test1:1:12] c.s.r.s.SegmentRunner - Repair command 228 on segment 12 returned with state DONE
INFO   [2017-03-31 18:57:41,202] [test1:1] c.s.r.s.RepairRunner - Repair amount done 12
INFO   [2017-03-31 18:57:41,202] [test1:1] c.s.r.s.RepairRunner - Repairs for repair run #1 done
```
Log from Cassandra node:
```
DEBUG [AntiEntropyStage:1] 2017-03-31 18:55:32,963 RepairMessageVerbHandler.java:144 - Got anticompaction request AnticompactionRequest{parentRepairSession=9d508810-1643-11e7-b728-5bfc5ef6abd0} org.apache.cassandra.repair.messages.AnticompactionRequest@cc41c373
INFO  [AntiEntropyStage:1] 2017-03-31 18:55:32,963 ActiveRepairService.java:439 - [repair #9d508810-1643-11e7-b728-5bfc5ef6abd0] Not a global repair, will not do anticompaction
```

for i in *Data.db; do sstablemetadata $i|grep Repa; done
```
Repaired at: 0
Repaired at: 0
Repaired at: 0
Repaired at: 0
Repaired at: 0
Repaired at: 0
Repaired at: 0
Repaired at: 0
```

**Comment by alekiv** [comment]
Could anybody confirm that reaper doesn't support incremental repairs?

**Comment by adejanovski** [comment]
Hi @alekiv,

Reaper does support incremental repair but I introduced a bug when I removed the use of deprecated repair methods in 2.2+.

Current master should work correctly now and perform anticompactions.

Could you please check and give me some feedback ?

Thanks

**Comment by adejanovski** [comment]
Hi @alekiv, 

any chance you could try current master and check that anticompaction works correctly ?

Thanks

**Comment by alekiv** [comment]
Hi @adejanovski,

just tested with current master and I can confirm that incremental repair works.

**Comment by adejanovski** [comment]
Hi @alekiv, 

thanks for the update.

> Although, I have question about "total_segments".
> With 12 nodes test cluster and incremental repair I see that total_segments=12.
> With non incremental repair total_segments=3072. I'm using vnodes (256 per node).
> 
> Is it expected behavior?
 

Incremental repair cannot be performed with subrange repair, and should only run once by node (and one node at a time). 
Reaper creates one fake segment per node to match the expected behavior.

TL;DR : yes, this is the expected behavior.

**Comment by alekiv** [comment]
@adejanovski 
Found it in other thread as well.
Thanks for clarification.
